modules/claims_api/lib/bgs_service/veteran_representative_service/read_all_veteran_representatives.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class VeteranRepresentativeService < ClaimsApi::LocalBGS
+    # type_code: form type (I.E. 21-22 vs 21-22A)
+    # ptcpnt_id: participant ID
+    def read_all_veteran_representatives(type_code:, ptcpnt_id:)
+      validate! type_code, ptcpnt_id
+
+      body = Nokogiri::XML::DocumentFragment.parse <<~EOXML
+        <ns0:CorpPtcpntIdFormTypeCode>
+          <veteranCorpPtcpntId>#{ptcpnt_id}</veteranCorpPtcpntId>
+          <formTypeCode>#{type_code}</formTypeCode>
+        </ns0:CorpPtcpntIdFormTypeCode>
+      EOXML
+      ret = make_request(action: 'readAllVeteranRepresentatives', body:)
+      ret&.dig('VeteranRepresentativeReturnList') || []
+    end
+
+    # bad idea?
```

Comment:
yea, curious what y'all's thoughts are on Threading lol

Reply:
Ya I think the threading looks straightforward.  I'm having a hard time finding any issues with it.

Reply:
@acovrig  Can you give me some background into the inclusion for this.  Is the idea that we would initially call this in place of calling `read_all_veteran_representatives` ?  Like a potentially more performant alternative to the `read_all_veteran_representatives` (and pulls both type codes?) ?

Reply:
yea @rockwellwindsor-va, IDK if we'd ever need to call 21-22 vs 21-22A separately, but figure it could be useful to have `read_all_veteran_representatives` available anywho. In the event that we need both (most of the time?) I figure a Thread would be nice so we don't have to wait for both, it'll make the call(s) at the same time. If we can do Threads, I feel like it could speed up requests.

Reply:
In the past I remember having some issues using threads and think that they are very heavy in Ruby. I'm pretty ignorant here, but I think Ruby has been developing stuff in the direction of lightweight concurrency with `Fiber`.